notion-core/src/main/scala/com/ambiata/notion/core/InputOutputLocations.scala

```diff
@@ -0,0 +1,106 @@
+package com.ambiata.notion.core
+
+import com.ambiata.mundane.io.{DirPath, FilePath, FileName}
+import org.apache.hadoop.fs.Path
+
+/**
+ * Type for execution locations.
+ *
+ * An execution can only take place locally or on Hdfs
+ */
+sealed trait ExecutionLocation {
+  /** the underlying location */
+  def location: Location
+
+  /** extract a Path for local and hdfs execution in order to pass it to a map-reduce job */
+  def path: Path
+
+  /** the underlying dirPath */
+  def dirPath: DirPath
+
+  /** the underlying filePath */
+  def filePath: FilePath
+}
+
+object ExecutionLocation {
+  def hdfsExecutionLocation(location: HdfsLocation): ExecutionLocation =
+    HdfsExecutionLocation(location)
+
+  def localExecutionLocation(location: LocalLocation): ExecutionLocation =
+    LocalExecutionLocation(location)
+}
+
+case class HdfsExecutionLocation(location: HdfsLocation) extends ExecutionLocation {
```

Comment:
No subtypes. Should carry the contents of the other constructor, not the constructor. I know this is harder before all @nhibberd's work gets unparked and finished - but we are working really hard to get rid of this and I don't want to add more.